### PR TITLE
Add an `Iterator.prototype.toObject` goody

### PR DESCRIPTION
**Pull request by miorel**
Like `Iterator.prototype.toMap` but converts to an object map rather than a `Map`.

### CODE REVIEW

File: tools/adventure-pack/goodies/typescript/Array.prototype.slidingWindows/index.test.ts [hunk 1]
```diff
@@ -6,11 +6,11 @@ describe("Array.prototype.slidingWindows", () => {
   it("can slide over an array", () => {
     const windows = [3, 1, 4, 1, 5, 9].slidingWindows(2);
 
-    expect([...windows.next().value!]).toEqual([3, 1]);
-    expect([...windows.next().value!]).toEqual([1, 4]);
-    expect([...windows.next().value!]).toEqual([4, 1]);
-    expect([...windows.next().value!]).toEqual([1, 5]);
-    expect([...windows.next().value!]).toEqual([5, 9]);
+    expect([...windows.next().value!]).toStrictEqual([3, 1]);
```
Comment (elimanzo) [inline tools/adventure-pack/goodies/typescript/Array.prototype.slidingWindows/index.test.ts:9]: What's the difference between going from `.toEqual` to `.toStrictEqual` is this similar to `==` and `===` from JS?

**Review by elimanzo — APPROVED** [review]
Looks good ! I had 1 question.

**Comment by miorel** [comment]
We should prefer `toStrictEqual` as it enforces more things. I had forgotten about it! https://ultimatecourses.com/blog/jest-toequal-tostrictequal-difference explains some of the details.

File: tools/adventure-pack/goodies/typescript/Array.prototype.slidingWindows/index.test.ts (hunk 1 above)
Comment (miorel) [reply tools/adventure-pack/goodies/typescript/Array.prototype.slidingWindows/index.test.ts:9]: We should prefer `toStrictEqual` as it enforces more things. I had forgotten about it! https://ultimatecourses.com/blog/jest-toequal-tostrictequal-difference explains some of the details.

**Pull request merged**